---
title: "WEB1 Notebook"
output:
  pdf_document: default
  html_notebook: default
---



```{r}
#task1
#1. Read the research article .
#2. Answer the following questions
#a. What is the medically relevant insight from the article?

The paper focuses primarily on white adipose tissue (WAT) in obese patients with selective insulin resistance. Post-obese patients are compared with controls who have never been obese, and whether there is a correlation between insulin pathways is analyzed. They find that some of these pathways remain unchanged, i.e., functional, while altered pathways are observed. However, with weight loss, some of these alterations are reversed.
By analyzing the 138 patient samples, selective insulin resistance can be observed. This is due to metabolic dysfunction and also contributes to the understanding of how type 1 diabetes works. This is a dysregulation of certain pathways that, when accumulated, contribute significantly to the progression of the disease.

#b. Which genomics technology/ technologies were used?

CAGE (cap analysis of gene expression): This helps map transcription start sites (TSS) at high resolution and quantitatively measure promoter activity. It was used to observe the regulation of specific promoters in WAT in obese and control patients. It was the baseline analysis for this study.

TFBS (Transcription Factor Binding Sites): This analysis was performed after CAGE to observe the transcription factors (TFs) responsible for changes in promoter activity of altered genes.

RNA-seq: This provides a global transcriptomic profile to identify genes whose expression is regulated by insulin, and is supported by the results obtained in CAGE.

qPCR: For the validation of genes of interest


#3. Further related research questions
#a. List and explain at least three questions/ hypotheses you can think of that extend the analysis presented in the paper.

1. Could unmapped WAT reads contain information about a microbiome altered by a virus or bacteria that could trigger insulin resistance?
  I think this is important since viruses or bacteria, such as E. coli, are present in different populations and cause metabolic alterations, and when chronic, some result in insulin resistance.
  Hypothesis: The presence of unmapped bacterial or viral microbiome reads in adipose tissue in obese individuals may be a factor in insulin resistance.

2. Could somatic copy number variations in genes accumulated due to obesity be a factor in interindividual heterogeneity in selective resistance?
  CNVs are large duplications or deletions in DNA that are spared in cancer samples and are not analyzed for other diseases. It would be interesting to observe the effects of CNVs in obese patients and whether they affect the insulin pathway.
  Hypothesis: Specific somatic copy number variations lead to insulin resistance at the adipose tissue level.

3. Insulin resistance is due to a subpopulation of functional and clonally expanded adipocytes in the adipose tissue ecosystem.
  Although the data in this paper are not single-cell data, one can search the database of another paper, such as (https://doi.org/10.1038/s41467-023-36983-2), and analyze the subpopulations of adipose tissues to identify adipocyte clusters and subpopulations.
  Hypothesis: A cellular ecosystem with a subpopulation of dysfunctional adipocytes is observed, causing selective insulin resistance at the tissue level.


#task2
Done
  URL github: git@github.com:sophiavelarde/RIKENCOURSE.git

#task3
Done
```

```{r}
#Task 4
#1. What is the square root of 10? 
I`m use the command for square

sqrt(10)

The square root of is 3.162278.
#2. What is the logarithm of 32 to the base 2?

I`m use command log but in added base 2 inside the operation

log(32,base=2)

Because 2^5 is 32 so the result is 5.

#3. What is the sum of the numbers from 1 to 1000?

Im use command sum but and use : because if im use , only take 1 and 1000

sum(1:1000)

the result is 500,500

#4. What is the sum of all even numbers from 2 to 1000? 

Im use 2 more commands seq and by. By because i need only pairs so this sum 2 and seq number generation

sum(seq(2,1000, by=2))
the result is 250,500

#5. How many pairwise comparisons are there for 100 genes?

choose(100,2)

There are 4950 possible pairwise comparisons between 100 genes.
This is because each pair represents a unique combination

#6. And how many ways to arrange 100 genes in triples?
 choose(100,3)
 
 There are 161,700 different ways to organize 100 genes into groups of three.

```
```{r}
#task5

#1. Use the R internal CO2 dataset (“data(CO2)”).

I´m use command data for call data CO2 and use command head because and validation if the data its correct.

data(CO2)
head(CO2)  

#2. Describe briefly the content of the CO2 dataset using the help function.

help(CO2)

when i´m use command help , the open helpand explain about the data, variables, details about data ,source and a some example abput code.

#Its a data about CO2 on the cold tolerance of the grass species Echinocloa crus-galli in different condition. They have 5 variables and 84 rows. This data was from Quebec and Mississipi.

#3. What is the average and median CO2 uptake of the plants from Quebec and Mississippi?

#charge dplyr
library(dplyr)

#calculate average and median for variable by plant type
CO2 %>%
  group_by(Type) %>%
  summarise(
    mean_uptake   = mean(uptake), #average
    median_uptake = median(uptake) #median
  )
Plant from Quebec showed high average and median CO2 uptake than Mississippi

#4. [Optional] In the “airway” example data from Bioconductor, how many genes are expressed in each sample? How many genes are not expressed in any sample?

#install package
install.packages("BiocManager")
BiocManager::install("airway")

#Airway data
library(airway)
data("airway")

#Obtain the count matrix
conteos <- assay(airway)
colSums(conteos > 0)

Each sample has about 23124 - 25998 expressed genes.

#no express
sum(rowSums(conteos) == 0)

It is observed that approximately 30208 genes are not expressed in any of the samples, which shows that the genes will be expressed depending on the biological conditions.

```


```{r}

#Task 6 - R Functions
#1. Write a function that calculates the ratio of the mean and the median of a given vector.

x <- function(x) mean(x) / median(x)

#2. Write a function that ignores the lowest and the highest value from a given vector and calculate the mean.

y <- function(x) mean(x[x != min(x) & x != max(x)])
This strategy is useful when we want to reduce the effect of outliers.

#3. Read about piping from here:https://r4ds.had.co.nz/pipes.html#pipes (you don’t have to learn everything, a basic understanding of the usage is enough). Write a short (max. 300 characters, no spaces) explanation of why, how, and when not to use pipes.

Pipe(%>%)makescodeeasiertoreadandunderstandwithoutcreatingintermediateobjectsornesting.Clearcodeisused,andverylongcodeorenvironment-dependentfunctionsshouldbeavoided.

#4. Familiarize yourself with the apply-family of functions (apply, lapply, sapply etc.) Write a short explanation (max. 300 characters, no spaces) of why they could be useful in your work.

Thefamily-applyallowsyoutosimplifyoperationsthatrequiremultiplearrays,lists,orvectorssothattheyareshorterandcanbeprocessedmorequickly,makingthemmoreefficientthanloops.
```



```{r}
#tasks7

#install packages

install.packages("remotes")
library(remotes)
install_url("http://emotion.utu.fi/wp-content/uploads/2019/11/nummenmaa_1.0.tar.gz", dependencies = TRUE)

library(ggplot2)
library(dplyr)

#ID

file("magic_guys.csv")

#Archive
data <- read.csv("/Users/sofiavelarde/Desktop/magic_guys.csv")
head(data)

Im charge this pack because I´m need this for design plot and histogram, and dowload mi data in R and work with this.

#1. Compare the distributions of the body heights of the two species from the 'magic_guys.csv' dataset graphically 

# a. using the basic 'hist' function as well as ‘ggplot’ and ‘geom_histogram’ functions from the ggplot2 package. Optimize the plots for example by trying several different 'breaks'

str(data)
This help to know the name for my column because in one chance have error use the name in the question. so now i have correct names.

#ggplot2
a <- data %>%
  ggplot(aes(x = length, fill = species)) + #from axis and fills bars by species
  geom_histogram(alpha = 1, position = "identity", bins = 15) + #bins optimizes visualitation and gem_histogram draws the histogram
  labs(title = "Length distribution by species", x = "Length", y = "Count") +
  theme_minimal()
#show graphic
a

The histogram show lenght distribution for both species in one plot and overlapping , Jedi have more distribution and sith overlapping.

#histogram species=jedi
b <- hist(data$length[data$species == "jedi"],#filters only jedi
     main = "Length distribution - Jedi",
     xlab = "Length", breaks = 10, col = "green")#breaks control number of bins and col sets the bar color

#histogram species=sith
c <-hist(data$length[data$species == "sith"],#filters only sith
     main = "Length distribution - Sith",
     xlab = "Length", breaks = 10, col = "purple") #col is for select color

Jedi and Sith have different length distributions.It has variability in its distribution

#b. Do the same comparison as in a. but with boxplots. If you want to use the ggplot2-package, use the functions ‘ggplot’ and ‘geom_boxplot’.

d <- data %>%
  ggplot(aes(x = species, y = length, fill = species)) + #fill=species colors boxes differently
  geom_boxplot(alpha = 1) + #geom_boxplot show median,quartile and outliers
  labs(title = "Length distribution by species",
       x = "Species", y = "Length") +
  theme_minimal()
#show graphic
d

Boxplots show medians, range, and variability.Jedi might have higher median length but less spread.

#c. Save the plots with the ‘png’, ‘pdf’, and ‘svg’ formats. In which situation would you use which file format?

Each file type depends on its purpose:
PNG: for presentations or websites
PDF: for printing the image
SVG: for uploading to another application for editing and when code support like Inkscape or Illustrator is needed.

#ggplot save
ggsave("a.png", a)
ggsave("a.pdf", a)
ggsave("a.svg", a)

ggsave("d.png", d)
ggsave("d.pdf", d)
ggsave("d.svg", d)



#histogram save
#b
png("b.png")               
hist(data$length[data$species=="jedi"],
     main="jedi", xlab="Length", breaks=10, col="green")
dev.off()    
png("b.pdf")               
hist(data$length[data$species=="jedi"],
     main="jedi", xlab="Length", breaks=10, col="green")
dev.off()  
png("b.svg")               
hist(data$length[data$species=="jedi"],
     main="jedi", xlab="Length", breaks=10, col="green")
dev.off()  

#c
pdf("c.png")               
hist(data$length[data$species=="sith"],
     main="sith", xlab="Length", breaks=10, col="purple")
dev.off()

pdf("c.pdf")               
hist(data$length[data$species=="sith"],
     main="sith", xlab="Length", breaks=10, col="purple")
dev.off()

pdf("c.svg")               
hist(data$length[data$species=="sith"],
     main="sith", xlab="Length", breaks=10, col="purple")
dev.off()

Histogram, unlike ggplot, the graph is created at the moment, so if you want to create and save it, you must create an ecosystem and close it with the dev.off command.

```
 

```{r}
#Task 8
#1. Install the Tidybiology package, which includes the data ‘chromosome’ and ‘proteins’

#Install packages from github
install.packages("devtools")

#install tidybiology from github
devtools::install_github("hirscheylab/tidybiology")

#load package
library(tidybiology)
library(tidyverse)
#load dataset
data("chromosome")
data("proteins")
#validation with first rows
head(chromosome)
head(proteins)
#a. Extract summary statistics (mean, median and maximum) for the following variables from the ‘chromosome’ data: variations, protein coding genes, and miRNAs. Utilize the tidyverse functions to make this as simply as possible.

#check column name
colnames(chromosome)

#calculate mean,median and max 
chromosome %>%
  summarise(
    across(c(variations, protein_codinggenes, mi_rna),#across applies functions to selected variables
           list(mean = mean, median = median, max = max),
           na.rm = TRUE)#ignore missing values
  )

Variations
Conclusion: The distribution is skewed; some chromosomes have many more variants than most.

Protein-coding genes
Conclusion: Longer chromosomes tend to have more protein-coding genes. Possible correlation with chromosome length.

miRNAs
Conclusion: The number of miRNAs also increases with chromosome length, although the variation is not as extreme as in variations.

#b. How does the chromosome size distribute? Plot a graph that helps to visualize this by using ggplot2 package functions.

#load package
library(ggplot2)
#check column name
colnames(chromosome)

ggplot(simplechromosome, aes(x = basepairs)) +
  geom_histogram(bins = 10, fill = "pink", color = "black")

The graph shows a scattered distribution, so it is not a normal distribution; it must be multimodal.

#c. Does the number of protein coding genes or miRNAs correlate with the length of the chromosome? Make two separate plots to visualize these relationships.

#protein vs coding genes
ggplot(chromosome, aes(x = basepairs, y = protein_codinggenes)) +
  geom_point() +
  geom_smooth(method = "lm")

#protein vs mirna
ggplot(chromosome, aes(x = basepairs, y = mi_rna)) +
  geom_point() +
  geom_smooth(method = "lm")

Both protein-coding genes and miRNAs are positively correlated, meaning that chromosome size influences them.


#d. Calculate the same summary statistics for the ‘proteins’ data variables length and mass. Create a meaningful visualization of the relationship between these two variables by utilizing the ggplot2 package functions. Play with the colors, theme- and other visualization parameters to create a plot that pleases you.

#load package
library(dplyr)

# basic
summary(proteins[,c("length","mass")])
plot(proteins$length, proteins$mass)

#with color
proteins %>%
  ggplot(aes(x = length, y = mass)) +
  geom_point(color = "green") +
  geom_smooth(method = "lm", color = "black") +
  theme_minimal()

It is observed that the length of the protein mass has a positive correlation
```


```

